Python/Product/Cookiecutter/Model/CookiecutterClientProvider.cs

```diff
@@ -38,11 +38,21 @@ static class CookiecutterClientProvider {
 
         private static CookiecutterPythonInterpreter FindCompatibleInterpreter() {
             var interpreters = PythonRegistrySearch.PerformDefaultSearch();
-            var compatible = interpreters
+            var all = interpreters
                 .Where(x => File.Exists(x.Configuration.InterpreterPath))
-                .OrderByDescending(x => x.Configuration.Version)
-                .FirstOrDefault(x => x.Configuration.Version >= new Version(3, 3));
-            return compatible != null ? new CookiecutterPythonInterpreter(compatible.Configuration.InterpreterPath) : null;
+                .Where(x => x.Configuration.Version >= new Version(3, 3))
+                .OrderByDescending(x => x.Configuration.Version);
```

Comment:
`all` got enumerated twice. You can call `ToList()` or `ToArray()` at the end to avoid it.